src/nbp_error.rs

```diff
@@ -0,0 +1,83 @@
+use serde::{Deserialize, Serialize};
+use std::fmt::{Display, Formatter};
+
+pub type NbpResult<T> = Result<T, NbpError>;
+
+/// Error type for NBP API operations.
+///
+/// Represents various types of errors that can occur when interacting
+/// with the NBP API, including network errors, parsing errors, and
+/// API-specific errors.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct NbpError {
+    r#type: ErrorType,
+    context: String,
+}
```

Comment:
This is a great case to use [thiserror](https://docs.rs/thiserror/latest/thiserror/) to add additional context to error type enum, as it typically would come from reqwest's http error of its own.

Reply:
You're right `thiserror` and `serde_json` do reduce boilerplate, but adding them as dependencies might be overkill if we're targeting multiple platforms or keeping the binary lean. For now, we can proceed with `thiserror` since it’s lightweight, but we should reassess if it becomes a constraint later (e.g., for no-std or WASM targets).

Reply:
Perfect!